packages/cosmic-swingset/tools/inquisitor.mjs

```diff
@@ -137,7 +148,50 @@ export const makeHelpers = ({ db, EV }) => {
     } else {
       console.warn('Warning: Unprefixed searches can be slow');
     }
-    return lazy ? sql.iterate(args) : sql.all(args);
+    return /** @type {any} */ (lazy ? sql.iterate(args) : sql.all(args));
+  };
+
+  // We need a RegExp.escape polyfill at least until our minimum supported
+  // Node.js includes it natively, however this is harmless to keep even then.
+  // It leaves leading ASCII alphanumerics untouched for readability and uses
+  // `\u00##` rather than `\x##` for simplicity, neither of which impact
+  // `scanVatStore`.
+  const needsEscape =
+    /([\^$\\.*+?()[\]{}|/])|[-,=<>#&!%:;@~''``""\s\uD800-\uDFFF]/gu;
+  const controlEscapes = {
+    '\t': '\\t',
+    '\n': '\\n',
+    '\v': '\\v',
+    '\f': '\\f',
+    '\r': '\\r',
+  };
+  const escapeChar = (ch, simple) => {
+    if (simple) return `\\${simple}`;
+    if (Object.hasOwn(controlEscapes, ch)) return controlEscapes[ch];
+    return ch
+      .split('')
+      .map(cu => `\\u${cu.charCodeAt(0).toString(16).padStart(4, '0')}`)
+      .join('');
+  };
+  const RegExpEscape = str => str.replace(needsEscape, escapeChar);
+
+  /**
+   * Return entries of a vat's KV store that mention `ref` as a distinct word.
+   * `ref` may be a vref such as "o-123", but may also be a value such as "123"
+   * or "v123" (that last case being particularly useful for finding a vat admin
+   * node in vatAdmin's KV store).
+   *
+   * @param {string} vatID
+   * @param {string} ref
+   */
+  const scanVatStore = (vatID, ref) => {
+    const candidates = kvGlob(`${vatID}.vs.*`, `*${ref}*`, true);
+    const re = RegExp(String.raw`\b(?:${RegExpEscape(ref)})\b`);
```

Comment:
It took me a minute to realize why the value glob is not sufficient. Can you add a comment clarifying that glob results need to be narrowed by word boundaries. I know that's literally what the RegExp does, but reading prose is a little easier.

Btw, why wrap the escaped regexp in a non capturing group ?